src/tiled/wangfiller.cpp

```diff
@@ -248,9 +250,9 @@ const Cell &WangFiller::getCell(const TileLayer &back,
                                 const QRegion &fillRegion,
                                 QPoint point) const
 {
-    if (!fillRegion.contains(point) && back.contains(point)) {
+    if (!fillRegion.contains(point) && (mIsInfinite || back.contains(point))) {
         return back.cellAt(point);
-    } else if (front.contains(point.x() - front.x(), point.y() - front.y())) {
+    } else if ((mIsInfinite || front.contains(point.x() - front.x(), point.y() - front.y()))) {
```

Comment:
This check can actually be dropped entirely, so we can change this to `else` and drop the current `else` part.